storage/inmem/inmem.go

```diff
@@ -94,13 +101,18 @@ type store struct {
 	rmu      sync.RWMutex                      // reader-writer lock
 	wmu      sync.Mutex                        // writer lock
 	xid      uint64                            // last generated transaction id
-	data     map[string]interface{}            // raw data
+	data     interface{}                       // raw or AST data
 	policies map[string][]byte                 // raw policies
 	triggers map[*handle]storage.TriggerConfig // registered triggers
 
 	// roundTripOnWrite, if true, means that every call to Write round trips the
 	// data through JSON before adding the data to the store. Defaults to true.
 	roundTripOnWrite bool
+
+	// returnASTValuesOnRead, if true, means that the store will eagerly convert data to AST values,
+	// and return them on Read.
+	// FIXME: naming(?)
```

Comment:
Seems ok to me.

Reply:
Maybe. To reiterate my previous point. Since what we're actually doing is eagerly converting to AST on _write_, this is a bit of a misnomer. We could consider this to be an implementation detail, but then we're kinda hiding the actual purpose behind this feature, which is to eliminate conversion cost on read.

Reply:
`returnASTValuesOnRead` is doing what it says so it seems like an implementation detail.